Documentation for new size unit system (#6255)
visgl/deck.gl: docs/developer-guide/coordinate-systems.md

In most deck.gl layers, every data object is expected to contain one or more **positions** (e.g. the center of a point, or the start and end of an arc, the vertices of a polygon, etc). Positions are expected to be supplied as two or three element arrays (`[x, y]` or `[x, y, z]`). Objects can also have **dimensions** (e.g. the radius of a circle, the width of a path, etc), most of which are specified as a single number. Positions and dimensions are used to construct a 3D world with your data.

> deck.gl supports a selection of coordinate systems, addressing diverse geospatial and non-geospatial [use cases](#world-space). Each layer can set its own `coordinateSystem` prop to specify how its positions should be interpreted. Layers using different coordinate systems can be composed together, which is very useful when dealing with datasets from different sources.

Reviewer: I'm a little surprised the `use cases` link points to `#world-space`. Is that intentional?

    {position: [33.22, 109.87, 1.455]}, // meter offsets from the coordinate origin
    ...
  ],
  radiusPixels: 2,
  sizeUnits: 'pixels'
})
```

## Concepts

Throughout this document you will find reference to the following **coordinate spaces** in deck.gl's projection system:

##### World space

The natural coordinate system of a dataset. It is usually determined by the source of the data, like the device it is generated on, and the format that it is stored as on disk or a cloud data warehouse.

Some examples of different world space definitions include a [Shapefile](https://en.wikipedia.org/wiki/Shapefile) with any [standard geospatial reference](https://epsg.io/); WGS84 as used by GeoJSON or KML; GPS traces; LIDAR scans; info-vis values that do not map to real-world dimensions; [Slippy Map](https://wiki.openstreetmap.org/wiki/Slippy_Map) tiles from services like OpenStreetMaps and Mapbox; 3D tiling standards such as [Cesium 3D Tiles](https://www.ogc.org/standards/3DTiles) and [Esri I3S](https://www.ogc.org/standards/i3s).

> Positions in different world spaces come with different scales and orientations, may be measured relative to static or dynamic reference points, and in many cases are not linear. It is common for an application to overlay multiple datasets from different world spaces into the same 3D view.

Reviewer: ```suggestion
Positions in different world spaces come with different scales and orientations, may be measured relative to static or dynamic reference points, and in many cases their scales are not linear. It is common for an application to overlay multiple datasets from different world spaces into the same 3D view.
```
Not sure what the best wording is, but the existing `in many cases are not linear` seemed to be referencing `Positions` which didn't seem right to me.

##### Common space

> To correctly compose data from various world spaces together, deck.gl transforms them into common space - an unified, intermediate 3D space that is a [right-handed Cartesian coordinate system](https://en.wikipedia.org/wiki/Cartesian_coordinate_system#In_three_dimensions). Once positions are in the common space, it is safe to add, substract, rotate, scale and extrude them as 3D vectors using standard linear algebra. This is the basis of all geometry processing in deck.gl layers.

Reviewer: ```suggestion
To correctly compose data from various world spaces together, deck.gl transforms them into common space - a unified, intermediate 3D space that is a [right-handed Cartesian coordinate system](https://en.wikipedia.org/wiki/Cartesian_coordinate_system#In_three_dimensions). Once positions are in the common space, it is safe to add, subtract, rotate, scale and extrude them as 3D vectors using standard linear algebra. This is the basis of all geometry processing in deck.gl layers.
```

> The transformation between the world space and the common space is referred to in deck.gl documentation as "project" and "unproject", a process controlled by both the specification of the world space, such as WGS84, and the [projection mode](/docs/developer-guide/views.md), such as Web Mercator. Projections are implemented as part of deck.gl's core.

Reviewer: ```suggestion
The transformation between the world space and the common space is referred to in deck.gl documentation as "project" (world space to common space) and "unproject" (common space to world space), a process controlled by both the specification of the world space, such as WGS84, and the [projection mode](/docs/developer-guide/views.md), such as Web Mercator. Projections are implemented as part of deck.gl's core.
```
Not sure if it's helpful here to be more explicit about project/unproject. I always forget which is which.

##### Screen space

> Top-left coordinate system runs from `[0, 0]` to `[viewportWidth, viewportHeight]`, measured in pixels.

Reviewer: ```suggestion
A top-left coordinate system that runs from `[0, 0]` to `[viewportWidth, viewportHeight]`, measured in pixels.
```

| `COORDINATE_SYSTEM.METER_OFFSETS`   | `[Δx, Δy, Δz]`   | `[longitude, latitude, altitude]` | Positions are given in meter offsets from a reference geo-location that is specified by `coordinateOrigin`. The `x` axis points map east, the `y` axis points map north, and `z` points up. Only works with geospatial views. |
| `COORDINATE_SYSTEM.LNGLAT_OFFSETS`    | `[Δlongitude, Δlatitude, Δaltitude]`   | `[longitude, latitude, altitude]` | Positions are given in degree offsets from a reference geo-location that is specified by `coordinateOrigin`. Only works with geospatial views. |
| `COORDINATE_SYSTEM.CARTESIAN`         | `[x, y, z]` | `[x, y, z]` (optional, default `[0, 0, 0]`) | A linear system that measures equally on all 3 axes. |

Remarks:

* Although [Universal Transverse Mercator](https://en.wikipedia.org/wiki/Universal_Transverse_Mercator_coordinate_system) uses similar notions as the `METER_OFFSETS` mode, be aware that the deck.gl offset system does not have the sophistication of the UTM spec and should not be used interchangeably. See the [limitations](#limitations-of-the-offset-system) section for details.

> * The `CARTESIAN` mode describes positions that are identical in the world space and the common space. It is the default coordinate system when rendering into non-geopatial views. When combined with geospatial views, the positions are treated as common space coordinates for that particular projection mode. The latter can be seen used by the [MVTLayer](/docs/api-reference/geo-layers/mvt-layer.md), where the data decoded from the tiles are already pre-projected onto the Web Mercator plane.

Reviewer: ```suggestion
* The `CARTESIAN` mode describes positions that are identical in the world space and the common space. It is the default coordinate system when rendering into non-geospatial views. When combined with geospatial views, the positions are treated as common space coordinates for that particular projection mode. The latter can be seen used by the [MVTLayer](/docs/api-reference/geo-layers/mvt-layer.md), where the data decoded from the tiles are already pre-projected onto the Web Mercator plane.
```
